Add attest protobuf

nick-mobilecoin opened the pull request.

nick-mobilecoin committed:
Add attest protobuf

nick-mobilecoin commented:
Current dependencies on/for this PR:
* main
  * **PR #2** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/probufs/2" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/probufs/2?utm_source=stack-comment).

nick-mobilecoin commented on .github/workflows/ci.yaml:
```diff
@@ -7,6 +7,13 @@ on:
   pull_request:
 
 jobs:
+  buf-lint:
```
buf-breaking doesn't work until we have proto files in the repo :(
So once this lands then I can add the buf-breaking ci check

nick-mobilecoin commented on attestapis/buf.md:
```diff
@@ -0,0 +1,38 @@
+# AttestAPIs
+
+The APIs required to interact with the `AttestService`.
```
I'm up for suggestions on these docs.

Also you can see the rendering [here](https://github.com/mobilecoinfoundation/probufs/blob/nick/attest-proto/attestapis/buf.md)

nick-mobilecoin commented on attestapis/buf.yaml:
```diff
@@ -0,0 +1,10 @@
+version: v1
+name: buf.build/nick-mc/attestapis
```
we'll need to decide where we publish to.

This PR shouldn't be merged until this is agreed on

nick-mobilecoin replied:
updated name

jcape reviewed (changes requested):
We should split types and services, I dislike "attestapis" as a name, and also I'd prefer to have the package end up being called `mobilecoinfoundation.attestation.v1` or something similar.

nick-mobilecoin committed:
Rename package to `mobilecoinfoundation.attestation`

Also:
- buf.build module has been renamed from `attestapis` to `foundation`
- Services has been broken out to it's own proto file

nick-mobilecoin commented on foundation/mobilecoinfoundation/attestation/v1/attest.proto:
```diff
@@ -0,0 +1,132 @@
+syntax = "proto3";
+
+// The APIs required to interact with the `AttestService`.
```
per https://docs.buf.build/bsr/documentation#package-documentation, package documentation goes in the alphabetically first module prior to the package directive

nick-mobilecoin commented on foundation/buf.lock:
```diff
@@ -0,0 +1,7 @@
+# Generated by buf. DO NOT EDIT.
```
I believe the `foundation` was the repo/module name that was agreed to go into, https://buf.build/mobilecoin
unless we create a new `mobilecoinfoundation` owner

nick-mobilecoin replied:
resolved we're creating a mobilecoinfoundation owner

nick-mobilecoin committed:
Remove one level of directory nesting

Since the idea is to have a `mobilecoinfoundation` package limiting this
repo to one package for now. If we decide later that more packages need
to exist we can move it down a directory

awygle reviewed (approved):
This matches what we discussed in the meeting.

jcape commented on mobilecoinfoundation/attestation/v1/attest.proto:
````diff
@@ -0,0 +1,132 @@
+// ```mermaid
+// sequenceDiagram
+//     autonumber
+//     Enclave_1->>+Enclave_2: MutualAttestRequest
````
```suggestion
//     Enclave_1->>+Enclave_2: MutualAttestRequest
//     Enclave_2->>Enclave_2: Verify MutualAttestRequest contents
```

nick-mobilecoin replied:
Updated, also fixed spelling of `Messge` on line 37

jcape reviewed (approved):
LGTM, one comment

nick-mobilecoin committed:
Add Enclave_2 verification to readme
